packages/react-reconciler/src/ReactFiberTreeReflection.js

```diff
@@ -275,10 +277,12 @@ export function findCurrentHostFiber(parent: Fiber): Fiber | null {
 
 function findCurrentHostFiberImpl(node: Fiber) {
   // Next we'll drill down this component to find the first HostComponent/Text.
+  const tag = node.tag;
   if (
-    node.tag === HostComponent ||
-    node.tag === HostText ||
-    (enableFloat ? node.tag === HostResource : false)
+    tag === HostComponent ||
+    tag === HostText ||
+    (enableFloat ? tag === HostResource : false) ||
+    (enableHostSingletons && tag === HostSingleton)
```

Comment:
Interesting use of two different stylistic patterns. :)

Reply:
Should this check `supportsSingletons` too?

Reply:
It was a litmus test to see how closely you reviewed these PRs, you passed!

No actually it was was rebase artifact. I tried the new style to see if CC could figure out how to omit the `|| !1` when the flag is false but no luck, either form leaves behind these artifacts in the stable prod build. I did it one way originally in HostSingleton, tried the other way in Float, and then missed this on rebase to main

Reply:
Also, re: `supportsSingletons` this file gets built without a hostconfig in one entrypoint so I can't import from that file or would need to make a fake entrypoint for it. Presumably it is safe enough to not have the check b/c there should be no fibers with those types so it's a little extra checking if you build with those flags on for a renderer that doesn't support them.

You did mention last time that this file was going to be deprecated at some point. let me know if you think it needs to be resolved